src/components/Seat/JetsSeat.integration.test.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { MockJetsContextProvider } from '../../__mocks__/MockJetsContext';
+
+import { paramsData, seatDataFirst } from './__fixtures__';
+import { JetsSeat } from './index';
+
+const setup = ({ data = {}, config = {}, params = {}, events = {} } = {}) => ({
+  user: userEvent.setup(),
+  ...render(
+    <MockJetsContextProvider config={config} events={events} params={{ ...paramsData(params) }}>
+      <JetsSeat data={data} />
+    </MockJetsContextProvider>
+  ),
+});
+
+describe('JetsSeat', () => {
+  describe('when any seat type is rendered', () => {
+    it('should trigger onClick handlers when clicked', async () => {
+      const onSeatClick = jest.fn();
+
+      const { user } = setup({
+        data: seatDataFirst(),
+        events: { onSeatClick },
+      });
+
+      await user.click(screen.getByText(/1A/));
+      expect(onSeatClick).toHaveBeenCalledTimes(1);
+      expect(onSeatClick).toHaveBeenCalledWith(seatDataFirst(), expect.any(Object), expect.any(Object));
+    });
+  });
+
+  describe('when hover mode is turned on', () => {
+    it('should trigger onMouseEnter and onMouseLeave handlers', async () => {
```

Comment:
What about 'should trigger onMouseEnter and onMouseLeave handlers when the seat is hovered and unhovered'

Reply:
Could we add another test to validate that the onMouseEnter and onMouseLeave are not called if the seat is hovered and unhovered and the hover mode is turned off please?

Reply:
I've added the test to the top section as that's the default render/

Reply:
Great, thanks!